cli/command/swarm/unlock.go

```diff
@@ -24,6 +26,14 @@ func newUnlockCommand(dockerCli *command.DockerCli) *cobra.Command {
 			client := dockerCli.Client()
 			ctx := context.Background()
 
+			// first see if the swarm is actually locked by running a node LS first, and bail
+			// if we don't get an error.  If we get any error, such as the swarm manager being
+			// locked, proceed anyway
+			_, err := client.NodeList(ctx, types.NodeListOptions{})
+			if err == nil {
+				return errors.New("Error: swarm is not locked")
+			}
```

Comment:
I feel it would be better to call `Info` and check that `LocalNodeState == LocalNodeStateLocked`. It's a bit more precise this way, and having a call to the info endpoint show up in the logs is marginally better than having the node list endpoint show up.